internal/tools/changes/metadata_test.go

```diff
@@ -0,0 +1,402 @@
+
+	newChange := &Change{
+		ID:          changeID,
+		Module:      "test/module",
+		Type:        FeatureType,
+		Description: "test description",
+	}
+
+	err := m.AddChange(newChange)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	found := false
+	for _, c := range m.Changes {
+		if c.ID == changeID {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		t.Errorf("Expected to find a change with id %s, but didn't", changeID)
+	}
+
+	m2 := getMetadata(t)
+
+	c, err := m2.GetChangeById("test-change-123456")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assertChangeEqual(t, newChange, c)
+
+	if c.SchemaVersion != SchemaVersion {
+		t.Errorf("Expected SchemaVersion %s, got %s", SchemaVersion, c.SchemaVersion)
+	}
+
+	err = m.RemoveChangeById("test-change-123456")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestMetadata_AddChanges(t *testing.T) {
+	m := getMetadata(t)
+
+	changes := []*Change{
+		&Change{
+			ID: "test-change-1",
+		},
+		&Change{
+			ID: "test-change-2",
+		},
+	}
+
+	err := m.AddChanges(changes)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, c := range changes {
+		_, err := m.GetChangeById(c.ID)
+		if err != nil {
+			t.Error(err)
+		}
+
+		err = m.RemoveChangeById(c.ID)
+		if err != nil {
+			t.Error(err)
+		}
+	}
+}
+
+func TestMetadata_CreateRelease(t *testing.T) {
+	m := getMetadata(t)
+	m.Changes = getMockChanges(t)
+
+	release, err := m.CreateRelease("test-release", map[string]VersionBump{
+		"test": {
+			From: "v1.0.0",
+			To:   "v1.0.1",
+		},
+		"other": {
+			From: "v1.3.2",
+			To:   "v1.4.0",
+		},
+	}, true)
+	if err != nil {
+		t.Fatalf("expected nil err, got %v", err)
+	}
+
+	if release.SchemaVersion != SchemaVersion {
+		t.Errorf("Expected SchmeVersion to be %s, got %s", SchemaVersion, release.SchemaVersion)
+	}
+
+	if len(release.Changes) != len(m.Changes) {
+		t.Errorf("Expected release to have %d changes, got %d", len(m.Changes), len(release.Changes))
+	}
+}
+
+func TestMetadata_GetChangeById(t *testing.T) {
+	m := getMetadata(t)
+	m.Changes = getMockChanges(t)
+
+	_, err := m.GetChangeById("invalid-id")
+	if err == nil {
+		t.Errorf("Expected non-nil err, got nil")
+	}
+
+	c, err := m.GetChangeById("test-feature-1")
+	if err != nil {
+		t.Fatalf("expected nil err, got %v", err)
+	}
+
+	if c.ID != "test-feature-1" {
+		t.Errorf("expected ID \"test-feature-1\", got %s", c.ID)
+	}
+}
+
+func TestMetadata_GetChanges(t *testing.T) {
+	m := getMetadata(t)
+	m.Changes = getMockChanges(t)
+
+	changes := m.GetChanges("other")
+	if len(changes) != 1 {
+		t.Fatalf("expected 1 Change, got %d", len(changes))
+	}
+
+	assertChangeEqual(t, m.Changes[3], changes[0])
+}
+
+func TestMetadata_AddChangesFromTemplate(t *testing.T) {
+	testCases := getTestTemplateCases(t)
+	m := getMetadata(t)
+
+	for id, tt := range testCases {
+		t.Run(id, func(t *testing.T) {
+			changes, err := m.AddChangesFromTemplate(tt.template)
+			if len(changes) == 0 {
+				if err == nil {
+					t.Errorf("expected non-nil err, got nil")
+				}
+			} else {
```

Comment:
If changes are expected or not probably should be annotated in the Test cases (`tt`)